adagrams/game.py

```diff
@@ -1,11 +1,125 @@
+
+import random
+
+
 def draw_letters():
-    pass
+    # array_of_strings = [ ]
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+    letter_pool = []
+    for letter, count in LETTER_POOL.items():
+        letter_pool.extend([letter] * count)
+    
+    hand = []
+    
+    for _ in range(10):
+        random_index = random.randint(0, len(letter_pool) - 1)
+        hand.append(letter_pool[random_index])
+        letter_pool.pop(random_index)
```

Comment:
This for loop is once again concise and well constructed! Great use of the pop method as well. The only consideration I would add is that the pop method does run in O(n) time. If you'd like an extra challenge, see if you can write this in a way that doesn't use the pop method and runs in O(1) time! Note: That code will be a little more involved but it will decrease the complexity.